### PR TITLE
justfile: use --locked to stick with Cargo.lock dependencies

### PR DESCRIPTION
**Pull request by djc**
Avoids breakage when semver-compatible dependency updates impose new constraints (such as MSRV).

See #2320.

cc @justahero

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -52,32 +52,32 @@ dnssec-ring: (default "--features=dnssec-ring" "--ignore=\\{async-std-resolver,h
...
 
 doc feature='':
-    cargo ws exec --ignore=hickory-dns cargo {{MSRV}} test --doc {{feature}}
+    cargo ws exec --ignore=hickory-dns cargo {{MSRV}} test --locked --doc {{feature}}
 
 # This tests compatibility with BIND9, TODO: support other feature sets besides openssl for tests
 compatibility: init-bind9
-    cargo test --manifest-path tests/compatibility-tests/Cargo.toml --all-targets --benches --examples --bins --tests --no-default-features --features=none;
-    cargo test --manifest-path tests/compatibility-tests/Cargo.toml --all-targets --benches --examples --bins --tests --no-default-features --features=bind;
+    cargo test --manifest-path tests/compatibility-tests/Cargo.toml --locked --all-targets --no-default-features --features=none;
```
Comment (bluejekyll) [inline justfile:71]: Similar comment as above.

**Review by bluejekyll — CHANGES_REQUESTED** [review]
Overall I think adding locked is fine, though in theory I would think MSRV changes shouldn’t be expected in minor patches? That seems like poor use of semver…

I guess we can rely on dependabot for ‘along sure we get new versions?

File: justfile [hunk 2]
```diff
@@ -52,32 +52,32 @@ dnssec-ring: (default "--features=dnssec-ring" "--ignore=\\{async-std-resolver,h
 
 # Run check on all projects in the workspace
 check feature='' ignore='':
-    cargo ws exec {{ignore}} cargo {{MSRV}} check --all-targets --benches --examples --bins --tests {{feature}}
-    cargo {{MSRV}} check --manifest-path fuzz/Cargo.toml --all-targets --benches --examples --bins --tests
+    cargo ws exec {{ignore}} cargo {{MSRV}} check --locked --all-targets {{feature}}
```
[the remark this answers is not in the document]
Comment (djc) [reply justfile:55]: Perhaps that had changed over time?

https://github.com/rust-lang/cargo/blob/ca330a26df8b10275a61a133822b7c7825b32e06/src/doc/man/includes/options-targets.md?plain=1#L52

**Comment by djc** [comment]
> Overall I think adding locked is fine, though in theory I would think MSRV changes shouldn’t be expected in minor patches? That seems like poor use of semver…

It is widely accepted practice to bump the MSRV in semver-compatible releases because the alternative is clearly worse, and this has been the case for a long time, so I'm not sure how you came to this conclusion.

**Comment by bluejekyll** [comment]
In patch releases or minor releases? Minor sure, but patch?

File: justfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (bluejekyll) [reply justfile:55]: Ok, maybe that is somewhat new?

**Review by bluejekyll — APPROVED** [review]

**Comment by djc** [comment]
> In patch releases or minor releases? Minor sure, but patch?

For 0.x versions staying semver-compatible means patch versions, so yes, that is still quite common.

**Pull request merged**